compiler/rustc_middle/src/mir/interpret/error.rs

```diff
@@ -241,17 +274,31 @@ pub struct ScalarSizeMismatch {
     pub data_size: u64,
 }
 
+macro_rules! impl_into_diagnostic_arg_through_debug {
+    ($($ty:ty),*$(,)?) => {$(
+        impl IntoDiagnosticArg for $ty {
+            fn into_diagnostic_arg(self) -> DiagnosticArgValue<'static> {
+                DiagnosticArgValue::Str(Cow::Owned(format!("{self:?}")))
+            }
+        }
+    )*}
+}
+
+impl_into_diagnostic_arg_through_debug! {
+    AllocId,
+    Pointer,
+    AllocRange,
+}
+
 /// Error information for when the program caused Undefined Behavior.
-pub enum UndefinedBehaviorInfo {
+#[derive(Debug)]
+pub enum UndefinedBehaviorInfo<'a> {
     /// Free-form case. Only for errors that are never caught!
-    Ub(String),
+    // Ub(String),
```

Comment:
Miri still needs this, I don't think we can get away with removing it. (Same for the other free-form errors here.)

Reply:
Oh, this moved to `Custom`. Not sure how I feel about grouping free-form errors and validity failures, those are quite different and I'd rather keep them separate.

Reply:
Yeah. I should probably keep this since not having this would require porting `miri` to translatable diagnostics. Which I think should still be nice to do, but just not in this PR now.

Reply:
Yeah certainly let's leave translating Miri for later. Also I think right now this would seriously slow down Miri development (looking at this PR and the enormous overhead introduced by translations) so I am not sure it would be a good idea.